plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImpl.java

```diff
@@ -300,8 +294,7 @@ private boolean isKubernetesServiceTemplateConfigured(DataCenter zone) {
                 LOGGER.warn(String.format("Unable to find the template %s to be used for provisioning Kubernetes cluster nodes", templateName));
                 return false;
             }
-            List<VMTemplateZoneVO> listZoneTemplate = templateZoneDao.listByZoneTemplate(zone.getId(), template.getId());
-            if (listZoneTemplate == null || listZoneTemplate.isEmpty()) {
+            if (CollectionUtils.isEmpty(templateJoinDao.newTemplateView(template, zone.getId(), true))) {
```

Comment:
The call `public List<TemplateJoinVO> newTemplateView(VirtualMachineTemplate template, long zoneId, boolean readyOnly)` as implemented in `TemplateJoinDaoImpl`, does a search including removed leading to a hit for a template that is not available. Is this intended?

Reply:
@DaanHoogland thanks for pointing it. I'll check that again

Reply:
any resolution on this @shwstppr ?

Reply:
@DaanHoogland I've checked this. `newTemplateView` is called here with `readyOnly=true` and deleted templates have `state=NULL`.